### PR TITLE
Thunder and module bumpup changes

### PR DESCRIPTION
**Pull request by HaseenaSainul**
Changes Include:
ThunderTools:---
Recipe update to take tools from ThunderTools repo
Thunder:----
[JSON] parsing updates
[Subsystem] location improvements
[WebSocket] update control frames based on mask bit 
[PrivilegedRequest]: handle multiple file descriptors
[SECURITY] Make sure the callsign is available even if the method does not report it
[PRIVILIGEDREQUEST] Add reference counting to the descriptors we pass
[PersistentPath] Ownership settings from thunder is removed
[Core] Forced TimeZone env update optional 
[FIXES] Fixes and performance optimizations and leaks
[IController] extended to generate json and COMRPC interfaces instead using hardcode implementation
[Hibernation] support added
ThunderInterface:---
[TimeSync] update time property usage as deprecated
[IBrowser] remove json:omit, since it is required for latest webkit browser
[VALUEPOINT] Move basic functionality into the ValuePoint reference implementatio
[SPLIT] Make sure the definitions get their own space
[REFREHSRATE] Add the refreshRate from resolution methods
ThunderNanoService:--
[BluetoothControl] Advertising/Discoverable modes
[Deinitialize] Remove Deinitialize call from the Initialization method, since it is already calling deactivation sequence
[COM-RPC] example stress test
[VERSION] Drop Version call if it is not used
[CLEANUP] Untangling the Wifi startup code
ThunderNanoServicesRDK:---
[MessageControl] clear lists only after deleting the outputDirectory
[MessageControl]Add the ModuelName to the Publishers of the MessageConntroller
[DeviceIdentification:AmLogic] Identifier fetching code added
[LocationSync] update based on IController interface changes 
ThunderClientLibraries:---
[OCDM adapter] Modified to fetch secbuf OCDM adapater from external and check mime type in protection metadata to detect encryption scheme
ThunderLibraries:----
[CMAKE] update to include PrevilagedRequest library also
Cobalt:---
Coablt2.23 version updates

### CODE REVIEW

File: recipes-wpe/wpeframework/include/wpeframework-plugins-rdk.inc [hunk 1]
```diff
@@ -28,7 +28,7 @@ PACKAGECONFIG ??= "\
     ${@bb.utils.contains('DISTRO_FEATURES', 'playready_vg', 'opencdmi_vgrdm', '', d)} \
     ${@bb.utils.contains('DISTRO_FEATURES', 'widevine', 'opencdmi_wv', '', d)} \
     ${@bb.utils.contains('DISTRO_FEATURES', 'security', 'securityagent', '', d)} \
-    deviceidentification deviceinfo locationsync monitor monitor_webkit tracing webkitbrowser \
+    deviceidentification deviceinfo locationsync monitor monitor_webkit messagecontrol webkitbrowser \
```
Comment (404ag) [inline recipes-wpe/wpeframework/include/wpeframework-plugins-rdk.inc:31]: Tracing module is still used by the integration. Please ensure that recipe is backward compatible

File: recipes-wpe/wpeframework/include/wpeframework-common.inc [hunk 2]
```diff
@@ -1,24 +0,0 @@
-# Common for thunder and modules which depends thunder
```
Comment (404ag) [inline recipes-wpe/wpeframework/include/wpeframework-common.inc:1]: What is the reason to remove the file?

File: recipes-wpe/wpeframework/wpeframework-ocdm-playready_git.bb [hunk 3]
```diff
@@ -1,10 +0,0 @@
-require include/wpeframework-ocdm-playready.inc
```
Comment (404ag) [inline recipes-wpe/wpeframework/wpeframework-ocdm-playready_git.bb:1]: What is the reason for remove this recipe?

File: recipes-wpe/wpeframework/wpeframework_git.bb [hunk 4]
```diff
@@ -1,8 +1,20 @@
+HOMEPAGE = "https://github.com/rdkcentral/Thunder"
 SUMMARY = "Web Platform for Embedded Framework"
 DESCRIPTION = "A C++ platform abstraction layer for generic functionality."
 
+SECTION = "wpe"
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=85bcfede74b96d9a58c6ea5d4b607e58"
+
+RECIPE_BRANCH ?= "development/cmake-tools-path-settings"
```
Comment (404ag) [inline recipes-wpe/wpeframework/wpeframework_git.bb:9]: Please do not bump to changes present only on the development branch.
If you need to use dev branch please use .bbappend on you platform layer.

**Review by 404ag — CHANGES_REQUESTED** [review]
Please split this PR into separate ones and separate fixes from bumps.

File: recipes-wpe/wpeframework/include/wpeframework-common.inc (hunk 2 above)
Comment (HaseenaSainul) [reply recipes-wpe/wpeframework/include/wpeframework-common.inc:1]: Now the thunderTools is moved to new repo, hence the same SRC details we cannot use for both thunder and tools now. Now all the required SRC settings are moved to corresponding recipe instead keeping in single place

File: recipes-wpe/wpeframework/wpeframework-ocdm-playready_git.bb (hunk 3 above)
Comment (HaseenaSainul) [reply recipes-wpe/wpeframework/wpeframework-ocdm-playready_git.bb:1]: We have two files for play ready 2.5 and 4.0 (wpeframework-ocdm-playready_2.5+git.bb & wpeframework-ocdm-playready_4.0+git.bb). So do we need to keep wpeframework-ocdm-playready_git.bb with same details of 4.0 ?

File: recipes-wpe/wpeframework/wpeframework_git.bb (hunk 4 above)
Comment (HaseenaSainul) [reply recipes-wpe/wpeframework/wpeframework_git.bb:9]: Sorry, I missed to update this. will correct

File: recipes-wpe/wpeframework/include/wpeframework-plugins-rdk.inc (hunk 1 above)
Comment (HaseenaSainul) [reply recipes-wpe/wpeframework/include/wpeframework-plugins-rdk.inc:31]: We can use either message control or tracing. I will add PACKAGE CONFIG to enable message control by default and tracing if message control disabled case

Comment (404ag) [reply recipes-wpe/wpeframework/include/wpeframework-plugins-rdk.inc:31]: Sounds like a plan. Please do so.

File: recipes-wpe/wpeframework/wpeframework_git.bb (hunk 4 above)
Comment (404ag) [reply recipes-wpe/wpeframework/wpeframework_git.bb:9]: Please correct that.

File: recipes-wpe/wpeframework/include/wpeframework-plugins-rdk.inc (hunk 1 above)
Comment (HaseenaSainul) [reply recipes-wpe/wpeframework/include/wpeframework-plugins-rdk.inc:33]: done

File: recipes-wpe/wpeframework/include/wpeframework-common.inc (hunk 2 above)
Comment (HaseenaSainul) [reply recipes-wpe/wpeframework/include/wpeframework-common.inc:1]: added

File: recipes-wpe/wpeframework/wpeframework_git.bb (hunk 4 above)
Comment (HaseenaSainul) [reply recipes-wpe/wpeframework/wpeframework_git.bb]: updated

File: recipes-extended/libcobalt/libcobalt_git.bb [hunk 5]
```diff
@@ -53,45 +58,33 @@ def get_cobalt_data_path(d):
         return "${WPEFRAMEWORK_DATA_PATH}/Cobalt"
 COBALT_DATA = "${@get_cobalt_data_path(d)}"
 
-do_configure() {
-    export PATH="$PATH:${S}/depot_tools"
+do_compile() {
```
Comment (404ag) [inline recipes-extended/libcobalt/libcobalt_git.bb:61]: Please split configure from the build here. On the build level, only ninja is necessary to execute. GN is needed only to generate the build configuration

File: recipes-extended/libcobalt/libcobalt_git.bb [hunk 6]
```diff
@@ -53,45 +58,33 @@ def get_cobalt_data_path(d):
...
-    export PATH="$PATH:${S}/../depot_tools"
-    if [ -f "${STAGING_DIR_TARGET}/usr/lib/libcobalt.so" ]
-    then
-        rm -rf "${STAGING_DIR_TARGET}/usr/lib/libcobalt.so"
-    fi
     install -d "${D}${libdir}"
-    install -m 0755 "${S}/src/out/${COBALT_PLATFORM_NAME}_${COBALT_BUILD_TYPE}/lib/libcobalt.so" "${D}${libdir}"
+    install -m 0755 "${S}/out/wpe/install/lib/libcobalt.so" "${D}${libdir}"
+    install -d "${D}${includedir}"/starboard
+    cp -prf "${S}"/starboard/*.h "${D}${includedir}"/starboard/
```
Comment (404ag) [inline recipes-extended/libcobalt/libcobalt_git.bb:82]: Please double quote `"${D}${includedir}"/starboard/`

File: recipes-extended/libcobalt/libcobalt_git.bb [hunk 7]
```diff
@@ -53,45 +58,33 @@ def get_cobalt_data_path(d):
...
+    install -d "${D}${includedir}"/third_party/starboard/wpe/shared
```
Comment (404ag) [inline recipes-extended/libcobalt/libcobalt_git.bb:83]: Please double quote the whole expression here and below

**Review by 404ag — CHANGES_REQUESTED** [review]

File: recipes-extended/libcobalt/libcobalt_git.bb (hunk 7 above)
Comment (HaseenaSainul) [reply recipes-extended/libcobalt/libcobalt_git.bb]: done

File: recipes-extended/libcobalt/libcobalt_git.bb (hunk 6 above)
Comment (HaseenaSainul) [reply recipes-extended/libcobalt/libcobalt_git.bb]: done

File: recipes-extended/libcobalt/libcobalt_git.bb (hunk 5 above)
Comment (HaseenaSainul) [reply recipes-extended/libcobalt/libcobalt_git.bb]: done

**Review by 404ag — APPROVED** [review]

**Review by 404ag — APPROVED** [review]

**Pull request merged**